sal-service/src/main/java/org/ow2/proactive/sal/service/service/PersistenceService.java

```diff
@@ -0,0 +1,277 @@
+        return cleanAllClustersFunction(sessionId);
+    }
+
+    /**
+     * Deregisters all edge devices by removing their entries.
+     * @param sessionId A valid session id
+     * @return true if all edge devices were deregistered successfully, false otherwise
+     */
+    public boolean cleanAllEdges(String sessionId) throws NotConnectedException {
+        LOGGER.info("Received cleanAllEdges endpoint call with sessionId: {}", sessionId);
+
+        // Check if the connection is active
+        if (!paGatewayService.isConnectionActive(sessionId)) {
+            LOGGER.warn("Session {} is not active. Aborting edge device cleanup.", sessionId);
+            throw new NotConnectedException();
+        }
+
+        // Perform the actual edge device cleanup
+        return cleanAllEdgeFunction(sessionId);
+    }
+
+    /**
+     * Helper function to perform cloud cleanup and return the result.
+     * @param sessionId A valid session id
+     * @return true if all clouds were cleaned successfully, false otherwise
+     */
+    public Boolean cleanAllCloudsFunction(String sessionId) {
+        LOGGER.info("Starting cloud cleanup function for sessionId: {}", sessionId);
+
+        try {
+            if (cloudService.isAnyAsyncNodeCandidatesProcessesInProgress(sessionId)) {
+                LOGGER.warn("Asynchronous node candidate retrieval is in progress. Cloud cleanup is deferred.");
+                return false;
+            }
+
+            // Retrieve all clouds
+            List<PACloud> allClouds = repositoryService.listPACloud();
+            if (allClouds.isEmpty()) {
+                LOGGER.warn("No clouds found to clean for sessionId: {}", sessionId);
+                return false;
+            }
+
+            // Collect all cloud IDs
+            final List<String> cloudIds = allClouds.stream().map(PACloud::getCloudId).collect(Collectors.toList());
+            LOGGER.info("Found {} clouds to clean for sessionId: {}", cloudIds.size(), sessionId);
+
+            // Perform cloud removal
+            boolean removeCloudsResult = cloudService.removeClouds(sessionId, cloudIds, true);
+            if (removeCloudsResult) {
+                LOGGER.info("Successfully removed all clouds for sessionId: {}", sessionId);
+            } else {
+                LOGGER.error("Failed to remove one or more clouds for sessionId: {}", sessionId);
+            }
+
+            return removeCloudsResult;
+
+        } catch (Exception e) {
+            // Log any errors with a message and stack trace
+            LOGGER.error("Unexpected error during cloud cleanup for sessionId: {}. Details: ", sessionId, e);
+            return false;
+        }
+    }
+
+    /**
+     * Helper function to perform cluster cleanup and return the result.
+     * @param sessionId A valid session id
+     * @return true if all clusters were undeployed successfully, false otherwise
+     */
+    public Boolean cleanAllClustersFunction(String sessionId) {
```

Comment:
Add @Transactional

Reply:
i did it in rest entry calls to encapsulate all micro functions in what is the endpoint call